tech/firewall-rule-format.md

```diff
@@ -575,15 +600,487 @@ For the record, the reason why the protocol is `*` in Azure is that
+      "protocol": "*",
+      "source_port_range": "*",
+      "destination_port_range": "*",
+      "source_address_prefix": "*",
+      "source_address_prefixes": [],
+      "destination_address_prefix": "*",
+      "destination_address_prefixes": [],
+      "source_port_ranges": [],
+      "destination_port_ranges": [],
+      "access": "Allow",
+      "priority": 100,
+      "direction": "Inbound",
+      "provisioning_state": "Succeeded",
+      "name": "Port_All",
+      "etag": "W/\"2873cc0e-f41e-4c94-a773-8b0982d0db55\""
+    }
+
+
+#### `destination_port_range` set to integer range
+
+Another way to allow access to all ports is to set *Destination port
+ranges* to `0-65535`.
+
+    {
+      "id": "/subscriptions/a8daa361-b951-46b5-8e9d-b375f2ffe9fd/resourceGroups/rg1/providers/Microsoft.Network/networkSecurityGroups/vm1-nsg/securityRules/Port_All",
+      "protocol": "*",
+      "source_port_range": "*",
+      "destination_port_range": "0-65535",
+      "source_address_prefix": "*",
+      "source_address_prefixes": [],
+      "destination_address_prefix": "*",
+      "destination_address_prefixes": [],
+      "source_port_ranges": [],
+      "destination_port_ranges": [],
+      "access": "Allow",
+      "priority": 100,
+      "direction": "Inbound",
+      "provisioning_state": "Succeeded",
+      "name": "Port_All",
+      "etag": "W/\"50b4dbec-2e8e-4675-b050-f87017d5118c\""
+    }
+
+
+#### `destination_port_ranges` set to multiple integer ranges
+
+If multiple port ranges are specified, they occur as a list value of
+the `destination_port_ranges` key as opposed to a single scalar value of
+the `destination_port_range` key. Note that this is very similar to
+`source_address_prefix` and `source_address_prefixes` that we saw
+earlier.
+
+For example, this security rule was created by setting *Destination port
+ranges* to `0-1024,1025-50000,50001-65535`.
+
+    {
+      "id": "/subscriptions/a8daa361-b951-46b5-8e9d-b375f2ffe9fd/resourceGroups/rg1/providers/Microsoft.Network/networkSecurityGroups/vm1-nsg/securityRules/Port_All",
+      "protocol": "*",
+      "source_port_range": "*",
+      "source_address_prefix": "*",
+      "source_address_prefixes": [],
+      "destination_address_prefix": "*",
+      "destination_address_prefixes": [],
+      "source_port_ranges": [],
+      "destination_port_ranges": [
+        "0-1024",
+        "1025-50000",
+        "50001-65535"
+      ],
+      "access": "Allow",
+      "priority": 110,
+      "direction": "Inbound",
+      "provisioning_state": "Succeeded",
+      "name": "Port_All",
+      "etag": "W/\"7f844086-b511-4fbb-b14e-67bb6d1635a6\""
+    }
+
+
+An interesting thing worth figuring out was if we could create a rule
+with *Destination port ranges* as something like `*,8080` to know if we
+we need to handle formats like these as well. An attempt to create a
```

Comment:
Single `we`.

Reply:
Fixed at [line 962](https://github.com/cloudmarker/guides/pull/5/files#diff-112a678346155cbc8647b65f47882715R962).